src/cascadia/TerminalSettingsModel/ApplicationState.cpp

```diff
@@ -9,8 +9,12 @@
 #include "ActionAndArgs.h"
 #include "JsonUtils.h"
 #include "FileUtils.h"
+#include "../../types/inc/utils.hpp"
 
 static constexpr std::wstring_view stateFileName{ L"state.json" };
+static constexpr std::wstring_view elevatedStateFileName{ L"elevated-state.json" };
+static constexpr std::wstring_view unelevatedStateFileName{ L"user-state.json" };

```

Comment:
Is there a reason you went with 3 files instead of 2?
Previously there was the idea that it would be sufficient to have a boolean flag of "forceUserState". This flag could be set to `true` for GeneratedProfiles, forcing elevated WT instances to always write that field into the unelevated `state.json`.

Of course this wouldn't need to be a boolean. We could also split the `MTSM_APPLICATION_STATE_FIELDS` macro into `MTSM_APPLICATION_STATE_FIELDS` and `MTSM_APPLICATION_STATE_FIELDS_ELEVATED`, allowing us to skip any boolean checks.

In either case I believe this would reduce the complexity of our code and state management.
The remaining code in this PR looks fine (aka ready to approve), but I think we might take a bit to think about this, as later file migrations are always a pain in the 🍑, unlike code changes.

Reply:
You know, that's what I wanted to do originally, but ran into weird edge cases and couldn't get it just right. b279601 was close but it couldn't persist the window state to the unelevated file just right. I'll gibe it another go. I swear I figured this out trying to fall asleep last night, but now it's gone 🙃

Reply:
Yea okay I remember what the trick was. When serializing from an elevated terminal, if we omit the window state from the blob we're going to write back to `state.json`, then the elevated window will blow away the window state currently saved in `state.json`.

I suppose that could be mitigated by first loading the `state.json` file to a json blob, then serializing the shared props on top of that blob, then writing that blob back. Kinda like what was in #11083 🤣